pulumi/src/backupCron.ts

```diff
@@ -0,0 +1,110 @@
+import { ServiceConfig, StatefulService } from ".";
+import * as k8s from '@pulumi/kubernetes'
+
+export const deployStsBackupCron = (asset: string, sts: StatefulService, namespace: string, provider: k8s.Provider) => {    
+  if(!sts.backup){
+    sts.backup = {
+      backupCount: 1,
+      // weekly 4 am on a Sunday
+      schedule: "0 4 * * 0"
+    }
+  }
+
+  const backupContainer = createBackupContainer(asset, namespace, sts)
+  const serviceAccountName = createRbac(asset, namespace, provider)
+  
+  createCronJob(asset, namespace, serviceAccountName, sts, backupContainer, provider)
+}
+
+const createCronJob = (asset: string, namespace: string, serviceAccountName: string, sts: StatefulService, backupContainer: k8s.types.input.core.v1.Container, provider: k8s.Provider) => {
+  new k8s.batch.v1.CronJob(`${asset}-backup-job`, {
+    metadata: {
+      name: `${asset}-backup-job`,
+      namespace: namespace,
+      annotations: { 'pulumi.com/skipAwait': 'true' },
+    },
+    spec: {
+      successfulJobsHistoryLimit: 1,
+      failedJobsHistoryLimit: 1,
+      concurrencyPolicy: "Forbid",
+      schedule: sts.backup?.schedule!!,
+      jobTemplate: {
+        spec: {
+          template: {
+            spec: {
+              serviceAccountName: serviceAccountName,
+              containers: [backupContainer],
+              restartPolicy: "Never"
+            },
+          },
+        },
+      },
+    },
+  }, { provider })
+}
+
+const createBackupContainer = (asset: string, namespace: string, sts: StatefulService): k8s.types.input.core.v1.Container => {
+  const pvcList = getPvcNames(asset, sts.replicas, sts.services)
+  return {
+    name: `${asset}-backup-runner`,
+    image: 'lukmyslinski/backuprunner:0.24',
+    args: ['-n', namespace, '-s', `${asset}-sts`, '-p', pvcList, '-r', `${sts.replicas}`, "-c", `${sts.backup?.backupCount}`],
+  }
+}
+
+const getPvcNames = (asset: string, replicas: number, services: ServiceConfig[]) => {
+  return (Array.from(Array(replicas).keys()).flatMap(n => {
+      return services.map(svc => `data-${svc.name}-${asset}-sts-${n}`)
+    }).filter((pvc) => pvc) as string[]).join(',')
+}
+
+
+const createRbac = (asset: string, namespace: string, provider: k8s.Provider) => {
```

Comment:
feels like we should be able to get the service account created by the snapshot controller for appropriate perms. could either pass back from cluster-launcher and assign as a stack reference output, or make that configurable  through cluster launcher and just pass that config down to stateful service -> backupCron

Reply:
This is the only point I'm not that sure of. I'd say it's better if we keep a single domain object defined in a single place/repo, otherwise doing any tweaks will require more back and forth. Had those been more tightly coupled I'd probably agree, but the cronjob is currently totally encapsulated here and I see more hard than good in splitting it up

Reply:
I can see your point to an extent about the encapsulation, but it also feels like we are missing out on leveraging a decent sized chunk of infra established by the helm chart. By exposing the service account through the stack output, it would guarantee that any changes made to the cluster launcher config would be associated via stack reference creating and "automated" link in a sense. Want to have a quick chat in discord about this just to prove both side out a bit further?

Reply:
Tbh I gave it another thought and yeah, let's just pull it up. It's a service account, shouldn't be touched very often

Reply:
as discussed in chat, leaving as is due to namespaced service accounts not allowing us to leverage `unchained-infra` role bindings/service accounts